aten/src/ATen/native/DilatedConvolution.cpp

```diff
@@ -0,0 +1,1090 @@
+
+#include <ATen/div_rtn.h>
+#include <algorithm>
+#include <tuple>
+#include "ATen/ATen.h"
+#include "ATen/native/im2col.h"
+#include "ATen/native/vol2col.h"
+#include "TH/THBlasUtils.h"
+
+#define TORCH_CHECK_DIM_SIZE(T, DIM, DIM_SIZE, SIZE) \
+  TORCH_CHECK(                                       \
+      T.dim() == DIM && T.size(DIM_SIZE) == SIZE,    \
+      "Need " #T " of dimension ",                   \
+      DIM,                                           \
+      " and " #T ".size[",                           \
+      DIM_SIZE,                                      \
+      "] == ",                                       \
+      SIZE,                                          \
+      " but got input to be of shape ",              \
+      T.sizes())
+
+/*
+  The following CALL... macros are for convenience of this source file
+  only. These should be replaced with ATen native functions as soon as
+  col2im, im2col, gemm, and gemv are ported.
+ */
```

Comment:
I'm... not going to tell you to change it, but I am curious about why you decided to use macros here. Most of these macros are only used two or three times, and the there isn't that much of a code size reduction. There are a lot of downsides to macros (e.g., they thwart automatic refactoring tools) and so my personal preference is to only use macros if I *absolutely* have to, and I don't really see that standard being met here. (And you're passing in most of the arguments in explicitly; yes, there are some arguments that are passed in implicitly like `n` in `CALLGEMM`, but that's pretty confusing for code readers, no?)

Reply:
The `CALLGEMM`, `CALLIM2COL`, etc macros are now eliminated.